src/bilinearinterpolationframe.cpp

```diff
@@ -275,18 +289,38 @@ void BilinearInterpolationFrame::sliceFrame() {
 
   // Create a boot::gli view of memory in downsample_bytes
   boost::gil::rgba8c_view_t gil = boost::gil::interleaved_view(
-      (size_t)frameWidth_, (size_t)frameHeight_,
-      (const boost::gil::rgba8c_pixel_t *)downsample_bytes,
+      frameWidth_, frameHeight_,
+      (const boost::gil::rgba8c_pixel_t *)raw_bytes.get(),
       frameWidth_ * sizeof(uint32_t));
 
   // Create gli::image to copy bytes into
   boost::gil::rgb8_image_t exp(frameWidth_, frameHeight_);
   boost::gil::rgb8_view_t rgbView = view(exp);
   boost::gil::copy_pixels(gil, rgbView);
-  free(
-      downsample_bytes);  // Free downsampled bytes. Image now in boost::gli obj
   // Compress memory (RAW, jpeg, or jpeg2000)
   data_ = compressor_->compress(rgbView, &size_);
+  if (!store_raw_bytes_) {
+    raw_compressed_bytes_ = NULL;
+    raw_compressed_bytes_size_ = 0;
+  } else {
+    // Convert memory buffer back from DICOM representation
+    // to standard representation
+    for (size_t idx = 0; idx < frame_mem_size; ++idx) {
+      const uint32_t color = raw_bytes[idx];
+      const uint32_t red = color & 0xFF;
+      // const uint32_t green = (color >> 8) & 0xFF;
+      const uint32_t  blue = (color >> 16) & 0xFF;
+      // const uint32_t  alpha = (color >> 24) & 0xFF;
+      // format alpha, red, green blue
+      raw_bytes[idx] = (color & 0xff00ff00) | (red << 16) | blue;
```

Comment:
Why aren't green and alpha included here?

Reply:
Oh, I think I get it, you're just swapping red and blue since alpha and green are already in the right place.

Reply:
The position of them isn't changed.  So I just masked and passed them through.  (color & 0xff00ff00) = Alpha 00 Green 00.  The position of red and blue are swapped.

Reply:
In general I'm a bit confused on why we're converting the pixel array to ABGR in bilinearinterpolationframe.cpp, but not nearestneighborframe.cpp (as far as I can tell).

Reply:
Good question.  Your correct.  It has to do with where the memory copy is being made in the code path.   In the nearest neighbor we copy the memory after the downsampling operation but before its converted from ARGB to ABGR. In the bilinearinterpolation the downsampling operation also does the ARGB to ABGR conversion.  The code here then makes a copy of the downsampled buffer and converts the pixels back to a standard.  I guess, if you thought there was value to it I could separate the downsampling and pixel conversion.  So add an additional loop which performs the pixel byte swaping.  If we did this we could remove the need to do this here. At the cost of a little more work it would make the code look more like the nearest neighbor code path and save us from the byte swapping here.